[primer] rename Router/REED to Mesh Extender/Extender-Capable Device (#200)
openthread/ot-docs: site/en/guides/thread-primer/router-selection.md

<a href="../images/ot-primer-cds.png"><img src="../images/ot-primer-cds.png" width="350" border="0" alt="OT Connected Dominating Set" /></a><figcaption style="text-align: center"><i>Example of a Connected Dominating Set</i></figcaption>
</figure>

Active Mesh Extenders must form a Connected Dominating Set (CDS), which means:

1.  There is an Active Mesh Extender-only path between any two Active Mesh Extenders.
1.  Any one Active Mesh Extender in a Thread network can reach any other Active Mesh Extender by staying
    entirely within the set of Active Mesh Extenders.
1.  Every End Device in a Thread network is directly connected to an Active Mesh Extender.

A distributed algorithm maintains the CDS, which ensures a minimum level of
redundancy. Every device initially attaches to the network as an End Device
(Child). As the state of the Thread network changes, the algorithm adds or
removes Active Mesh Extenders to maintain the CDS.

Thread adds Active Mesh Extenders to:

*   Increase coverage if the network is below the Active Mesh Extender threshold of 16
*   Increase path diversity
*   Maintain a minimum level of redundancy
*   Extend connectivity and support more Children

> Thread removes Active Mesh Extenders to:

Reviewer: ```suggestion
Thread removes Active Mesh Extenders (switching them to Standby Mesh Extender) to:
```

*   Reduce the Routing state below the maximum of 32 Active Mesh Extenders
*   Allow new Active Mesh Extenders in other parts of the network when needed

## Upgrade to an Active Mesh Extender

> After attaching to a Thread network, the Child device may elect to

Reviewer: ```suggestion
After attaching to a Thread network, a Child device that is a (Standby) Mesh Extender may elect to
```

become an Active Mesh Extender. Before initiating the MLE Link
Request process, the Child sends an Address Solicit message to the
Leader, asking for a Router ID. If the Leader accepts, it responds

> with a Router ID and the Child upgrades itself to an Active Mesh

Reviewer: ```suggestion
with a Router ID and the Child upgrades itself from a Standby Mesh Extender to an Active Mesh
```

In some scenarios, it may be necessary to establish a one-way receive link.

After an Active Mesh Extender reset, neighboring Active Mesh Extenders
may still have a valid receive link with the reset Active Mesh
Extender. In this case, the reset Active Mesh Extender sends a Link

> Request message to re-establish the Active Mesh Extender link.

Reviewer: ```suggestion
Request message to re-establish the existing valid link.
```
(this looks a bit clearer - refers to the link already defined above.)

An End Device may also wish to establish a receive link with
neighboring non-Parent Active Mesh Extenders to improve multicast
reliability. We'll learn more about this when we get to Multicast
Routing.

## Recap

What you've learned:

*   Active Mesh Extenders in a Thread network must form a Connected Dominating Set (CDS)

> *   Thread devices are upgraded to Active Mesh Extenders or downgraded to Standby Mesh Extenders to

Reviewer: ```suggestion
*   Thread devices (of the Mesh Extender device type) can be upgraded to Active Mesh Extenders or downgraded to Standby Mesh Extenders to
```

    <div>Which of these rules are not enforced by a Connected Dominating Set
    (CDS)?</div>
    <div>
      <div>There is an Active Mesh Extender-only path between any two Active Mesh Extenders.</div>
      <div>Incorrect.</div>
    </div>
    <div>
      <div>Any one Active Mesh Extender in a Thread network can reach any other Active Mesh Extender by
        staying entirely within the set of Active Mesh Extenders.</div>
      <div>Incorrect.</div>
    </div>
    <div>
      <div>Every End Device in a Thread network is directly connected to an
        Active Mesh Extender.</div>
      <div>Incorrect.</div>
    </div>
    <div correct>
      <div>Only one Active Mesh Extender in a Thread network may be a Border Router.</div>
      <div>Correct. A Thread network may have multiple Border Routers.</div>
    </div>
  </devsite-multiple-choice>
</div>

<div>
  <devsite-multiple-choice>

>     <div>Why might an Active Mesh Extender be removed from a Thread network?</div>

Reviewer: "removed" as a term looks incorrect here. The Mesh Extender is downgrade from Active to Standby in this case, not removed?

Maybe:

```suggestion
    <div>Why might an Active Mesh Extender be removed (i.e. downgraded to Standby Mesh Extender) from a Thread network?</div>
```

    <div correct>

>       <div>To reduce the Routing state below the maximum of 32 Active Mesh Extenders.</div>

Reviewer: ```suggestion
      <div>To keep the Routing state below the maximum of 32 Active Mesh Extenders.</div>
```

Correction here. "To reduce" sounds like it was above 32 already. "Keep" better suggests it was already <= 32 and now we want to go to <= 31 to guarantee that we stay <= 32. (And still have room for an extra one later maybe.)

<div>
  <devsite-multiple-choice>
    <div>What must happen before a Standby Mesh Extender that is attempting to become
    an Active Mesh Extender can establish direct links with the other Active Mesh Extenders?</div>
    <div correct>
      <div>The Standby Mesh Extender must send an Address Solicit message to the network Leader.
      </div>
      <div>Correct.</div>
    </div>
    <div correct>
      <div>The Leader must grant a Router ID to the Standby Mesh Extender.</div>

>       <div>Correct. Without a Router ID, the Standby Mesh Extender remains a Child device.</div>

Reviewer: ```suggestion
      <div>Correct. Without a Router ID, the Standby Mesh Extender remains an End Device (Child).</div>
```
